BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -0,0 +1,1037 @@
+        {
+            while (true)
+            {
+                lastUpdate = Time.time;
+                UpdateStatus();
+                yield return PilotLogic();
+            }
+        }
+
+        protected override void OnGUI()
+        {
+            base.OnGUI();
+
+            if (!pilotEnabled || !vessel.isActiveVessel) return;
+
+            if (!BDArmorySettings.DEBUG_LINES) return;
+            GUIUtils.DrawLineBetweenWorldPositions(vesselTransform.position, debugPosition, 5, Color.red); // The point we're asked to fly to
+
+            // Vel vectors
+            GUIUtils.DrawLineBetweenWorldPositions(vesselTransform.position, vesselTransform.position + Vector3.Project(vessel.Velocity(), Vector3.ProjectOnPlane(vesselTransform.up, upDir)).normalized * 10f, 2, Color.cyan); //forward/rev
+            GUIUtils.DrawLineBetweenWorldPositions(vesselTransform.position, vesselTransform.position + Vector3.Project(vessel.Velocity(), Vector3.ProjectOnPlane(vesselTransform.right, upDir)).normalized * 10f, 3, Color.yellow); //lateral
+
+            GUIUtils.DrawLineBetweenWorldPositions(vesselTransform.position, vesselTransform.position + vesselTransform.up * 1000, 3, Color.white);
+            GUIUtils.DrawLineBetweenWorldPositions(vesselTransform.position, vesselTransform.position + -vesselTransform.forward * 100, 3, Color.yellow);
+            GUIUtils.DrawLineBetweenWorldPositions(vesselTransform.position, vesselTransform.position + vessel.Velocity().normalized * 100, 3, Color.magenta);
+        }
+
+        #endregion events
+
+        #region Actual AI Pilot
+
+        protected override void AutoPilot(FlightCtrlState s)
+        {
+
+            upDir = VectorUtils.GetUpDirection(vesselTransform.position);
+            
+            
+            if (BDArmorySettings.DEBUG_TELEMETRY || BDArmorySettings.DEBUG_AI)
+            {
+                debugString.AppendLine($"Current Status: {currentStatus}");
+                debugString.AppendLine($"Has Propulsion: {hasPropulsion}");
+                debugString.AppendLine($"Has Weapons: {hasWeapons}");
+                if (targetVessel)
+                {
+                    Vector3 relVel = RelVel(vessel, targetVessel);
+                    float minRange = Mathf.Max(MinEngagementRange, targetVessel.GetRadius() + vesselStandoffDistance);
+                    debugString.AppendLine($"Target Vessel: {targetVessel.GetDisplayName()}");
+                    debugString.AppendLine($"Can Intercept: {CanInterceptShip(targetVessel)}");
+                    debugString.AppendLine($"Near Intercept: {NearIntercept(relVel, minRange)}");
+                    debugString.AppendLine($"Near Intercept Burn Time: {nearInterceptBurnTime:G3}");
+                    debugString.AppendLine($"Near Intercept Approach Time: {nearInterceptApproachTime:G3}");
+                    debugString.AppendLine($"Lateral Velocity: {lateralVelocity:G3}");
+                }
+            }
```

Comment:
I've also added in the following locally (which is what `lastStatusMode` is for)
```csharp
if (BDArmorySettings.DEBUG_AI)
{
    if (lastStatusMode != currentStatusMode)
    {
        Debug.Log("[BDArmory.BDModuleOrbitalAI]: Status of " + vessel.vesselName + " changed from " + lastStatusMode + " to " + currentStatus);
    }
    lastStatusMode = currentStatusMode;
}
```